File: app/src/main/java/io/github/fatimazza/fbmatchschedule/network/RetrofitHelper.java
package io.github.fatimazza.fbmatchschedule.network;

import io.github.fatimazza.fbmatchschedule.BuildConfig;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitHelper {

    private static RetrofitHelper retrofitHelper;

    private TheSportDBApi sportDBApi;

    private RetrofitHelper() {
        Retrofit mRetrofit = new Retrofit.Builder()
                .baseUrl(BuildConfig.BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .client(setupLoggingInterceptor().build())
                .build();
        sportDBApi = mRetrofit.create(TheSportDBApi.class);
    }

    public static RetrofitHelper getInstance() {
        if (retrofitHelper == null) {
            retrofitHelper = new RetrofitHelper();
        }
        return retrofitHelper;
    }

    public TheSportDBApi getMoviesServices() {
        return sportDBApi;
    }

    private OkHttpClient.Builder setupLoggingInterceptor() {
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BODY);

        OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
        httpClient.addInterceptor(logging);

        return httpClient;
    }

}

File: app/src/main/java/io/github/fatimazza/fbmatchschedule/network/RetrofitHelper.kt
package io.github.fatimazza.fbmatchschedule.network

import io.github.fatimazza.fbmatchschedule.BuildConfig
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitHelper private constructor() {

    val theSportDBApi: TheSportDBApi

    init {
        val mRetrofit = Retrofit.Builder()
                .baseUrl(BuildConfig.BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .client(setupLoggingInterceptor().build())
                .build()
        theSportDBApi = mRetrofit.create(TheSportDBApi::class.java)
    }

    private fun setupLoggingInterceptor(): OkHttpClient.Builder {
        val logging = HttpLoggingInterceptor()
        logging.level = HttpLoggingInterceptor.Level.BODY

        val httpClient = OkHttpClient.Builder()
        httpClient.addInterceptor(logging)

        return httpClient
    }

    companion object {

        private var retrofitHelper: RetrofitHelper? = null

        val instance: RetrofitHelper
            get() {
                if (retrofitHelper == null) {
                    retrofitHelper = RetrofitHelper()
                }
                return retrofitHelper as RetrofitHelper
            }
    }

}